### DISCUSSION TITLE
Tab Control -- how to change tab color and make background color the same

### DISCUSSION THREAD

**Discussion by sonicsmooth** [AvaloniaUI/Avalonia#10547] [General]
My Simple theme tab control with only minor Margin and Padding adjustments looks like this:
![image](https://user-images.githubusercontent.com/672429/222813144-03687b42-b4ff-4533-9285-98f1feb5db80.png)

How do I:
a) change the tab color
b) make the background of the main area the same color.

I think I need to mess with the template settings but I'm having a *major* brain block with how *any* of this works... 

I want it to look like the samples.
![image](https://user-images.githubusercontent.com/672429/222813371-a29fbd03-2f85-492a-96b4-b5791bfac6c7.png)

**Comment by j4587698** [comment]
Maybe you can check this website: https://docs.avaloniaui.net/docs/controls/tabcontrol

**Reply by sonicsmooth** [reply]
Interesting, I spent much of the day on Friday staring at the xaml source for tabcontrol, but I didn't see this link.  Thank you for posting.

**Reply by sonicsmooth** [reply]
`WrapPanel` looks like the main thing here.  I'll play around with this.  Thank you. :)

**Comment by timunie** [comment]
@sonicsmooth luckily Avalonia is open source, so you can look up the orginal styles including the ones used in the sample app. Please remember to switch to 0.10.18-branch if you are using 0.10.18. The master branch is for 11.0 only. 

If you have problems finding the correct files, please let me know the version you use and I'll point you.

**Reply by sonicsmooth** [reply]
Thank you for the response.   

I was looking at the xaml source much of Friday, and I think my question ultimately comes from seeing too many `contentpresenter`, `controltemplate`, `datatemplate`, `itemtemplate`, etc., and most of these properties just bind up to `{TemplateBinding WhateverProperty}`.  In the case of `TabControl`, it's made up of  `TabItem`, `TabStrip`, `TabStripItem`, etc., down to `TextBox` I think at the lower levels.  I'm not sure whether to modify stuff by doing `<Style Selector="TabControl /template/ ContentPresenter">` or by attempting to modify the template directly such as `<TabControl.ContentTemplate.../>` like here https://github.com/AvaloniaUI/Avalonia/discussions/5885.

I was trying start by looking where the color for the tab is defined.  The source says it's `ThemeForegroundBrush`.  I grepped the entire source tree and didn't find this name defined (I only found it used), so I'm not sure where the light blue color comes from.  I was able to change the color eventually with some type of background styling (I don't remember where at this exact moment), but then I wasn't sure how to tie that in with the background of the main content, as shown in the document linked elsewhere

In short, I don't have a good mental model for how to interpret what I see in the source and therefore how to manipulate it is still a matter of asking "how do I ...".

Thank for you the support on this forum.  Maybe I'll switch back to fpga this week :)

**Reply by timunie** [reply]
Again, which version of Avalonia do you use? This is important for me to know. I'll guide you tomorrow further.

**Reply by sonicsmooth** [reply]
I'm on 11.0.0-preview4.  I'll follow the details here https://docs.avaloniaui.net/docs/controls/tabcontrol.  I think `WrapPanel` looks like a key thing which I didn't notice before.  It's this type of syntax ` <Style Selector="TabItem:pointerover /template/ ContentPresenter#PART_ContentPresenter">` that I haven't figured out yet.  It all still looks very ad hoc to me, but clearly there is a pattern there which I've yet to see.

**Reply by timunie** [reply]
1. Read about Styles and Style selectors: https://docs.avaloniaui.net/docs/styling/selectors
2. Here's the original style:
  a) https://github.com/AvaloniaUI/Avalonia/blob/release/11.0.0-preview4/src/Avalonia.Themes.Simple/Controls/TabControl.xaml
  b) https://github.com/AvaloniaUI/Avalonia/tree/release/11.0.0-preview4/samples/SampleControls/HamburgerMenu

**Reply by timunie** [reply]
ah sorry I forgot, also mind these styles: 
- https://github.com/AvaloniaUI/Avalonia/blob/release/11.0.0-preview4/src/Avalonia.Themes.Simple/Controls/TabItem.xaml

**Comment by sonicsmooth** [comment]
Thanks for the help.  I was able to get my panel to looks like this.
I also had to add a color to make sure that even when not focused, the color stayed the same as selected.

![image](https://user-images.githubusercontent.com/672429/223557847-eccac23d-e85a-4996-9e10-c4ca7bb0b943.png)
